### PR TITLE
Feature/fill profile

### PR DESCRIPTION
**Pull request by adriankulinski**
User profile feature

### CODE REVIEW

**Comment by JanuszBartosz** [comment]
Beautiful code, I'm really impressed! :)

File: LoyaltyProgram_Intern/src/main/java/com/pgs/soft/service/impl/UserProfileServiceImpl.java [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+		this.userProfileRepository = userProfileRepository;
+		this.userRepository = userRepository;
+		this.hobbyRepository = hobbyRepository;
+	}
+
+	@Override
+	public void fill(UserProfileDTO userProfileDTO, String userEmail) {
+		User user = userRepository.findOneByEmail(userEmail).get();
+		UserProfile userProfile = user.getUserProfile();
+		userProfile.setName(userProfileDTO.getName());
```
Comment (kfirsztPgs) [inline LoyaltyProgram_Intern/src/main/java/com/pgs/soft/service/impl/UserProfileServiceImpl.java:40]: Please move mapping to seperate method.

File: LoyaltyProgram_Intern/src/main/java/com/pgs/soft/service/UserProfileService.java [hunk 2]
```diff
@@ -0,0 +1,8 @@
+package com.pgs.soft.service;
+
+import com.pgs.soft.dto.UserProfileDTO;
+
+public interface UserProfileService {
+
+	public void fill(UserProfileDTO userProfileDTO, String userEmail);
```
Comment (kfirsztPgs) [inline LoyaltyProgram_Intern/src/main/java/com/pgs/soft/service/UserProfileService.java]: Please change name to save.

File: LoyaltyProgram_Intern/src/main/java/com/pgs/soft/domain/Hobby.java [hunk 3]
```diff
@@ -0,0 +1,59 @@
...
+	
+	@Id
+	@GeneratedValue(strategy = GenerationType.IDENTITY)
+	private Integer id;
+	
+	@Column(name = "hobby_name")
+	private String hobbyName;
+	
+	@ManyToMany(mappedBy = "hobbies")
+	private Set<UserProfile> userProfiles;
```
Comment (kfirsztPgs) [inline LoyaltyProgram_Intern/src/main/java/com/pgs/soft/domain/Hobby.java]: Is is needed here?

File: LoyaltyProgram_Intern/src/main/java/com/pgs/soft/controller/UserController.java [hunk 4]
```diff
@@ -11,31 +17,35 @@
...
-		userService.register(userDTO);
-		return "Registered.";
+	public void initBinder(WebDataBinder binder){
+		binder.addValidators(userProfileValidator);
+	}
+
+
+	@RequestMapping(value = "/profile", method = RequestMethod.POST)
+	public String fillProfile(@Valid @RequestBody UserProfileDTO userProfileDTO, HttpServletRequest httpServletRequest){
+		HttpSession session = httpServletRequest.getSession();
```
Comment (kfirsztPgs) [inline LoyaltyProgram_Intern/src/main/java/com/pgs/soft/controller/UserController.java]: Retrieve logged user from Spring Security Context

File: LoyaltyProgram_Intern/src/main/java/com/pgs/soft/controller/RegisterController.java [hunk 5]
```diff
@@ -0,0 +1,41 @@
...
+
+import com.pgs.soft.RegisterValidator;
+import com.pgs.soft.dto.UserDTO;
+import com.pgs.soft.service.UserService;
+
+@Controller
+@ResponseBody
+public class RegisterController {
+	
+	UserService userService;
```
Comment (kfirsztPgs) [inline LoyaltyProgram_Intern/src/main/java/com/pgs/soft/controller/RegisterController.java]: Please autowire by field - as it is in onther places

File: LoyaltyProgram_Intern/src/main/java/com/pgs/soft/UserProfileValidator.java [hunk 6]
```diff
@@ -0,0 +1,53 @@
...
+		}
+	}
+	
+	private void validateSurname(Errors errors, UserProfileDTO userProfileDTO){
+		if(!userProfileDTO.getSurname().isEmpty() && !isStringLetter(userProfileDTO.getSurname())){
+			errors.reject("surname.incorrect","Surname is incorrect");
+		}
+	}
+	
+	private boolean isStringLetter (String name){
```
Comment (kfirsztPgs) [inline LoyaltyProgram_Intern/src/main/java/com/pgs/soft/UserProfileValidator.java]: Please move it to some Utils class

File: LoyaltyProgram_Intern/src/main/java/com/pgs/soft/UserProfileValidator.java [hunk 7]
```diff
@@ -0,0 +1,53 @@
...
+	}
+	
+	private void validateName(Errors errors, UserProfileDTO userProfileDTO){
+		if(!userProfileDTO.getName().isEmpty() && !isStringLetter(userProfileDTO.getName())){
+			errors.reject("name.incorrect","Name is incorrect");
+		}
+	}
+	
+	private void validateSurname(Errors errors, UserProfileDTO userProfileDTO){
+		if(!userProfileDTO.getSurname().isEmpty() && !isStringLetter(userProfileDTO.getSurname())){
```
Comment (kfirsztPgs) [inline LoyaltyProgram_Intern/src/main/java/com/pgs/soft/UserProfileValidator.java]: What if surname has '-' sign?

File: LoyaltyProgram_Intern/src/main/java/com/pgs/soft/UserProfileValidator.java [hunk 8]
```diff
@@ -0,0 +1,53 @@
...
+import org.springframework.validation.Errors;
+import org.springframework.validation.Validator;
+
+import com.pgs.soft.dto.UserProfileDTO;
+
+@Component
+public class UserProfileValidator implements Validator {
+
+	@Autowired
+	HttpSession session;
```
Comment (kfirsztPgs) [inline LoyaltyProgram_Intern/src/main/java/com/pgs/soft/UserProfileValidator.java]: Please use Spring Security Context to retrieve logged user

File: LoyaltyProgram_Intern/src/main/java/com/pgs/soft/RegisterValidator.java [hunk 9]
```diff
@@ -12,12 +12,12 @@
...
 
 @Component
-public class UserDtoValidator implements Validator {
+public class RegisterValidator implements Validator {
 	
 	private final UserService userService;
 	
 	@Autowired
-	public UserDtoValidator(UserService userService){
+	public RegisterValidator(UserService userService){
```
Comment (kfirsztPgs) [inline LoyaltyProgram_Intern/src/main/java/com/pgs/soft/RegisterValidator.java]: Please stay with convention - if we inject bean by field we should do this everywhere

File: LoyaltyProgram_Intern/src/main/java/com/pgs/soft/service/UserProfileService.java (hunk 2 above)
Comment (adriankulinski) [inline LoyaltyProgram_Intern/src/main/java/com/pgs/soft/service/UserProfileService.java]: Done.

File: LoyaltyProgram_Intern/src/main/java/com/pgs/soft/controller/UserController.java (hunk 4 above)
Comment (adriankulinski) [inline LoyaltyProgram_Intern/src/main/java/com/pgs/soft/controller/UserController.java]: Done.

File: LoyaltyProgram_Intern/src/main/java/com/pgs/soft/controller/RegisterController.java (hunk 5 above)
Comment (adriankulinski) [inline LoyaltyProgram_Intern/src/main/java/com/pgs/soft/controller/RegisterController.java]: Done.

File: LoyaltyProgram_Intern/src/main/java/com/pgs/soft/UserProfileValidator.java (hunk 8 above)
Comment (adriankulinski) [inline LoyaltyProgram_Intern/src/main/java/com/pgs/soft/UserProfileValidator.java]: Done.

File: LoyaltyProgram_Intern/src/main/java/com/pgs/soft/RegisterValidator.java (hunk 9 above)
Comment (adriankulinski) [inline LoyaltyProgram_Intern/src/main/java/com/pgs/soft/RegisterValidator.java]: Done.

File: LoyaltyProgram_Intern/src/main/java/com/pgs/soft/UserProfileValidator.java (hunk 7 above)
Comment (adriankulinski) [inline LoyaltyProgram_Intern/src/main/java/com/pgs/soft/UserProfileValidator.java]: Done, and first letter has to be upper case.

File: LoyaltyProgram_Intern/src/main/java/com/pgs/soft/UserProfileValidator.java (hunk 6 above)
Comment (adriankulinski) [inline LoyaltyProgram_Intern/src/main/java/com/pgs/soft/UserProfileValidator.java]: I've changed that class, please take a look.

File: LoyaltyProgram_Intern/src/main/java/com/pgs/soft/domain/Hobby.java (hunk 3 above)
Comment (adriankulinski) [inline LoyaltyProgram_Intern/src/main/java/com/pgs/soft/domain/Hobby.java]: What do you mean?

File: LoyaltyProgram_Intern/src/main/java/com/pgs/soft/service/impl/UserProfileServiceImpl.java (hunk 1 above)
Comment (adriankulinski) [inline LoyaltyProgram_Intern/src/main/java/com/pgs/soft/service/impl/UserProfileServiceImpl.java:40]: Done.

File: LoyaltyProgram_Intern/src/main/java/com/pgs/soft/domain/Hobby.java (hunk 3 above)
Comment (adriankulinski) [inline LoyaltyProgram_Intern/src/main/java/com/pgs/soft/domain/Hobby.java]: Ok, done.

**Pull request merged**